sfit.py

```diff
@@ -1,11 +1,36 @@
 import argparse
+import importlib
 
-def parse_args()->argparse.Namespace:
-  ...
+commands = ["init", "add", "commit", "branch", "checkout", "merge", "tag"]
 
-def main(args) -> int:
-  return 0
+def parse_args() -> argparse.Namespace:
+  # Top level parser
+  parser = argparse.ArgumentParser()
+
+  # Subparsers to parse subcommands
+  subparsers = parser.add_subparsers(title="subcommands", required=True)
+  
+  # Import command modules
+  # Each command has a module with the corresponding name in the commands directory
+  
+  # The module exposes an initialize function which takes the subparsers object and
+  # adds a subparser for the command it represents
+
+  # Also attaches the command handler function as func in the args namespace
+  # This allows the parser to directly call the correct function from the
+  # parsed args namespace in the main function
+  
+  for command in commands:
+    command_module = importlib.import_module("commands." + command)
+    command_module.initialize(subparsers)
 
+  return parser.parse_args()
+
+def main(args) -> int:
+  if hasattr(args, "func"):
```

Comment:
Feel like this is check is unnecessary, you should put a type hint on args in the func def

Reply:
The check is present for safety in case any command is missing implementation, or some different kind of parser is used in future.

`args` is of type `argparse.Namespace` which has attributes depending on the command. I will add the type hint but it won't be of much assistance since the attributes depend on the specific command and its arguments that were parsed.

Reply:
I think you should design the main parser so that parsing itself fails in such a case.
We can stick to argparse, so I'd highly recommend adding the type hint.

Reply:
On second thought, the check seems fine. Do add the type hint though